src/models/Factor.js

```diff
@@ -223,10 +223,18 @@ function (Okta, Q, factorUtil, Util, Errors, BaseLoginModel) {
         if (this.pushFactorHasAutoPush()) {
           data.autoPush = this.get('autoPush');
         }
-
+        // for FACTOR_REQUIRED with email magic link we dont need to show otp code input field,
+        // hence we dont need to post the otp code to the server
+        if (this.options.appState.get('isFactorRequired') && this.get('factorType') === 'email') {
```

Comment:
I think that instead of deleting the property we just set a few lines above, we can probably add some logic to not set it in the first place (this will go in line 219). Something like
```
...
else if (this.addPasscode()) {
  data.passCode = this.get('answer');
}
```